src/services/spellCheckerService.ts

```diff
@@ -56,21 +57,18 @@ class SpellCheckerServiceImpl {
     }
   }
 
-  private async makeApiRequest(words: string[]): Promise<SpellCheckResult> {
+  private async makeApiRequest(words: string[], languageCode: string): Promise<SpellCheckResult> {
     try {
-      const response = await fetch(this.endpoint, {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json;charset=utf-8',
-        },
-        body: JSON.stringify(words),
+      // Use Amplify's REST API client with the spellcheck API
+      const { response } = await post({
+        apiName: 'spellcheck',
+        path: `/${languageCode}/bulk-lookup`,
+        options: {
+          body: words
+        }
       });
-
-      if (!response.ok) {
-        throw new Error(`API request failed: ${response.status}`);
-      }
-
-      const result = await response.json();
+      const res = await response;
```

Comment:
You don't need to change this, but I get sticky about "half word variables" like `err` and `fmt`. _I_ would just call this `raw` or `rawResponse` just for the sake of readability. In this case this is pretty minor so don't feel like you need to change it, just be prepared to be nitpicked on this in the future ;)